index.js

```diff
@@ -263,13 +269,37 @@ module.exports = {
               __frameId: params.frameId,
               _initialPriority: request.initialPriority,
               _priority: request.initialPriority,
-              _initiator: params.initiator.url,
-              _initiator_line: params.initiator.lineNumber,
               pageref: currentPageId,
               request: req,
               time: 0
             };
 
+            switch (params.initiator.type) {
+              case 'script':
+                if (
+                  params.initiator.stack
+                  && params.initiator.stack.callFrames
+                  && params.initiator.stack.callFrames.length > 0
+                ) {
+                  const firstCallFrame = params.initiator.stack.callFrames[params.initiator.stack.callFrames.length - 1];
+                  entry._initiator = firstCallFrame.url;
+                  entry._initiator_line = firstCallFrame.lineNumber;
```

Comment:
Re-reading the docs I see that lineNumber is 0 based. To match what a human would expect when looking at the file, we should add 1. I notice WebPageTest uses the 0 based numbering, but I think it's ok to diverge from WPT in this case.

Reply:
It'd be awesome to add columnNumber here as well. WPT has that, with the name _initiator_column.

Reply:
WPT also has _initiator_function, which we could get from firstCallFrame .functionName.